letter-shop/index.md

```diff
@@ -0,0 +1,634 @@
+
+As soon as yur plugin connects to the ledger it will start listening for incoming transfers and should raise this event every time one occurs.
+
+Replace `//Handle incoming transfers...` with the following code:
+
+```js
+  plugin.on('incoming_prepare', function (transfer) {
+
+    if (transfer.amount !== '10') {
+      
+      //Transfer amount is incorrect
+      console.log(`    - Payment received for the wrong amount (${transfer.amount})... Rejected`)
+
+      plugin.rejectIncomingTransfer(transfer.id, {
+        code: 'F04',
+        name: 'Insufficient Destination Amount',
+        message: `Please send exactly 10 ${ledgerInfo.currencyCode}, you sent ${transfer.amount}`,
+        triggered_by: plugin.getAccount(),
+        triggered_at: new Date().toISOString(),
+        forwarded_by: [],
+        additional_info: {}
+      })
+
+    } else {
+
+      //Lookup fulfillment from condition attached to incoming transfer
+      const fulfillment = fulfillments[transfer.executionCondition]
+
+      if(!fulfillment) {
+
+        //We don't have a fulfillment for this condition
+        console.log(`    - Payment received with an unknwon condition: ${transfer.executionCondition}`)
+
+        plugin.rejectIncomingTransfer(transfer.id, {
+          code: 'F05',
+          name: 'Wrong Condition',
+          message: `Unable to fulfill the condition:  ${transfer.executionCondition}`,
+          triggered_by: plugin.getAccount(),
+          triggered_at: new Date().toISOString(),
+          forwarded_by: [],
+          additional_info: {}
+        })
+      }
+
+      console.log(` 4. Accepted payment with condition ${transfer.executionCondition}.`)
+      console.log(`    - Fulfilling transfer on the ledger using fulfillment: ${fulfillment}`)
+      
+      // The ledger will check if the fulfillment is correct and if it was submitted before the transfer's
+      // rollback timeout
+      plugin.fulfillCondition(transfer.id, fulfillment).catch(function() {
+        console.log(`    - Error fulfilling the transfer`)
+      })
+
+      console.log(`    - Payment complete`)
+    }
+  })
+```
+
+In this code we first check that the incoming transfer is for the correct amount, and if it's not we raise an ILP Error and call the `rejectIncomingTransfer()` function to reject the transfer. If there were other business rules we wanted to evaluate before accepting the transfer we could do these here too.
+
+If the transfer is correct, we extract the condition and use this to look up the fulfillment from our local store. If we can't find it then we aren't able to fufill the transfer so we reject it.
+
+> **NOTE:** There are other ways to map a fulfillment to an incoming transfer, including techniques where the fulfillment is derived from the transaction data. For an example of such a technique have a look at the [PSK protocol](https://interledger.org/rfcs/0016-pre-shared-key/).
```

Comment:
".. the PSK protocol, which will be introduced in the [Streaming Payments](./streaming-payments) tutorial."

Reply:
Will update when that is live